fetch/cosmwasm_contract/src/contract.rs

```diff
@@ -1,69 +1,398 @@
 use cosmwasm_std::{
-    to_binary, Api, Binary, Env, Extern, HandleResponse, InitResponse, Querier,
-    StdResult, Storage, HumanAddr
+    log,
+    to_binary,
+    Api,
+    BankMsg,
+    Binary,
+    CanonicalAddr,
+    Coin,
+    CosmosMsg,
+    Env,
+    Extern,
+    HandleResponse,
+    HandleResult,
+    HumanAddr,
+    InitResponse,
+    Querier,
+    StdError,
+    StdResult,
+    Storage, //ReadonlyStorage,
 };
 
-use crate::msg::{RoleResponse, HandleMsg, InitMsg, QueryMsg, U256};
-use crate::state::{config, /*config_read,*/ State};
+use crate::msg::{HandleMsg, InitMsg, QueryMsg, RoleResponse, Uint128};
+use crate::state::{config, config_read, State};
+
+/* ***************************************************
+ * **************    Initialization      *************
+ * ***************************************************/
 
 pub fn init<S: Storage, A: Api, Q: Querier>(
     deps: &mut Extern<S, A, Q>,
-    _env: Env,
-    _msg: InitMsg,
+    env: Env,
+    msg: InitMsg,
 ) -> StdResult<InitResponse> {
+    let current_block_number = env.block.height;
+
+    let mut paused_since_block = msg.paused_since_block.unwrap_or(u64::MAX);
+    if paused_since_block < current_block_number {
+        paused_since_block = current_block_number
+    }
+
+    let delete_protection_period = msg.delete_protection_period.unwrap_or(0u64);
+    let earliest_delete = current_block_number + delete_protection_period;
+
     let state = State {
+        supply: msg.deposit, // Uint128::zero(), // TMP(LR)
+        refunds_fees_accrued: Uint128::zero(),
         next_swap_id: 0,
+        sealed_reverse_swap_id: 0,
         relay_eon: 0,
+        upper_swap_limit: msg.upper_swap_limit,
+        lower_swap_limit: msg.lower_swap_limit,
+        cap: msg.cap,
+        swap_fee: msg.swap_fee,
+        paused_since_block,
+        earliest_delete,
```

Comment:
Just nitpicking: For now, I would comment-out all code which deals with contract deletion concept (wherever the `delete_protection_period` and `earliest_delete` is used).
Reason being that we don't know yet, wether cosmwasm app supports contract deletion(= its state, code, logs, etc. ...) on conceptual level.